$MirOS: ports/devel/cvs/patches/patch-lib_vasnprintf_c,v 1.1 2011/05/06 23:17:27 tg Exp $
--- lib/vasnprintf.c.orig	Mon May 23 17:44:11 2005
+++ lib/vasnprintf.c	Thu Jul 28 15:49:14 2011
@@ -51,6 +51,8 @@
 /* Checked size_t computations.  */
 #include "xsize.h"
 
+__RCSID("$MirOS$");
+
 /* Some systems, like OSF/1 4.0 and Woe32, don't have EOVERFLOW.  */
 #ifndef EOVERFLOW
 # define EOVERFLOW E2BIG
@@ -100,7 +102,12 @@ local_wcslen (const wchar_t *s)
 # define DIRECTIVE char_directive
 # define DIRECTIVES char_directives
 # define PRINTF_PARSE printf_parse
+#if 0
+/* disabled for security reasons, to avoid having %n in writable memory */
 # define USE_SNPRINTF (HAVE_DECL__SNPRINTF || HAVE_SNPRINTF)
+#else
+# define USE_SNPRINTF 0
+#endif
 # if HAVE_DECL__SNPRINTF
    /* Windows.  */
 #  define SNPRINTF _snprintf
@@ -591,11 +598,12 @@ VASNPRINTF (CHAR_T *resultbuf, size_t *l
 		  {
 		    size_t maxlen;
 		    int count;
-		    int retcount;
+#if USE_SNPRINTF
+		    int retcount = 0;
+#endif
 
 		    maxlen = allocated - length;
 		    count = -1;
-		    retcount = 0;
 
 #if USE_SNPRINTF
 # define SNPRINTF_BUF(arg) \
